src/components/app/App.js

```diff
@@ -1,28 +1,43 @@
 import React, { Component } from 'react';
 import Posters from '../posters/Posters';
 import MovieDetails from '../movie_details/MovieDetails';
-import movieData from '../../movieData';
+import ErrorCode from '../error_code/ErrorCode';
+import { getApiData } from '../../apiCalls';
 import './App.css';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
-      selectedMovie: null
+      movies: [],
+      errorCode: null,
+      selectedMovie: null,
+      fetchingMovie: null
```

Comment:
You could use this to track wether or not there are movies being fetched, but couldn't we do the same thing by checking on the `movies` piece of state? I would suggest a refactor to implement that - this will keep state nice and clean, and only filled with the _most_ necessary pieces of information.

Reply:
I was using this for conditional rendering in the render method of App since it's going to hit that first after updating state... essentially to say if an error is thrown when visiting homepage don't give them a back button option, but if it's thrown while fetching a movie give them a back button so they can attempt to open a different movie. But this is gone after utilizing Router and componentDidMount.